tests/Feature/ApplicantControllerTest.php

```diff
@@ -0,0 +1,140 @@
+            'veteran_status_id' => $updateData['veteran_status_id'],
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ];
+        $response->assertJsonFragment($expectedData);
+        $this->assertDatabaseHas(
+            'applicant_classifications',
+            $applicant->applicant_classifications->first()->toArray()
+        );
+
+        // Ensure that submitting applicant classifications deletes elements that were not resubmitted
+        // and creates new elements/updates old elements.
+        $oldApplicantClassification = $applicant->applicant_classifications->first()->toArray();
+        $updateData = [
+            'citizenship_declaration_id' => CitizenshipDeclaration::inRandomOrder()->first()->id,
+            'veteran_status_id' => VeteranStatus::inRandomOrder()->first()->id,
+            'applicant_classifications' => [
+                0 => $oldApplicantClassification,
+                1 => $newApplicantClassification,
+            ],
+        ];
+        $response = $this->actingAs($applicant->user)
+            ->putJson("$this->baseUrl/applicant/$applicant->id/profile", $updateData);
+        $response->assertOk();
+
+        $applicant->refresh();
+        $expectedData = [
+            'citizenship_declaration_id' => $updateData['citizenship_declaration_id'],
+            'veteran_status_id' => $updateData['veteran_status_id'],
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ];
+        $response->assertJsonFragment($expectedData);
+        foreach ($applicant->applicant_classifications->toArray() as $applicantClassification) {
+            $this->assertDatabaseHas(
+                'applicant_classifications',
+                $applicantClassification
+            );
+        }
+
+        // Ensure that submitting no applicant classifications deletes all
+        // previous applicant classifications attached to applicant.
+        $updateData = [
+            'citizenship_declaration_id' => CitizenshipDeclaration::inRandomOrder()->first()->id,
+            'veteran_status_id' => VeteranStatus::inRandomOrder()->first()->id,
+            'applicant_classifications' => [],
+        ];
+
+        $response = $this->actingAs($applicant->user)
+            ->putJson("$this->baseUrl/applicant/$applicant->id/profile", $updateData);
+        $response->assertOk();
+
+        $applicant->refresh();
+        $expectedData = [
+            'citizenship_declaration_id' => $updateData['citizenship_declaration_id'],
+            'veteran_status_id' => $updateData['veteran_status_id'],
+            'applicant_classifications' => [],
+        ];
+        $response->assertJsonFragment($expectedData);
+        $this->assertEmpty($applicant->applicant_classifications);
+        foreach ($applicant->applicant_classifications->toArray() as $applicantClassification) {
```

Comment:
`$applicant->applicant_classifications->toArray()` should be an empty array at this point, so the assertDatabaseMissing test will never run.